bugfix: S3C-3114 close inactive free sockets in http agent

jonathan-gramain opened the pull request:
Automatically close free sockets in the HTTP/HTTPS agent that have
been inactive for a certain time. The inactivity time is set to 55
seconds, which is lower than the configured 60 seconds timeout for
idle connections on bucketd server, avoiding the risk of reusing a
connection that is just being closed by the server, resulting in an
ECONNRESET error.

It uses the third-party npm module agentkeepalive which provides the
extra "freeSocketTimeout" agent option, not available in the regular
node.js http.Agent.

jonathan-gramain committed:
bugfix: S3C-3114 close inactive free sockets in http agent

Automatically close free sockets in the HTTP/HTTPS agent that have
been inactive for a certain time. The inactivity time is set to 55
seconds, which is lower than the configured 60 seconds timeout for
idle connections on bucketd server, avoiding the risk of reusing a
connection that is just being closed by the server, resulting in an
ECONNRESET error.

It uses the third-party npm module agentkeepalive which provides the
extra "freeSocketTimeout" agent option, not available in the regular
node.js http.Agent.

bert-e commented:
# Hello jonathan-gramain,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/bucketclient/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue S3C-3114 contains:


* `7.4.7`

* `7.7.1`

* `7.8.0`


Considering where you are trying to merge, I expected to find:


* `7.4.7`

* `7.7.1`

* `7.8.0`

* `8.0.0`

* `8.1.0`


Please check the `Fix Version/s` of S3C-3114, or the target
branch of this pull request.

jonathan-gramain committed:
bugfix: S3C-3114 update arsenal dep

Update arsenal dependency to fix a build issue related to 'joi'
module, as it is now imported as "@hapi/joi" in more recent arsenal
versions.

ilkescality commented on package.json:
```diff
@@ -26,14 +26,15 @@
     "eslint-config-scality": "scality/Guidelines#71a059ad",
-    "eslint-plugin-react": "4.2.3"
+    "eslint-plugin-react": "4.2.3",
+    "mocha": ""
```
What's reason to remove mocha version?

ilkescality replied:
nvm, for a sec I thought it was new, just realized it's just on new line.

jonathan-gramain replied:
yes, yarn did this on its own actually

miniscruff commented on lib/RESTClient.js:
```diff
@@ -4,10 +4,18 @@ const assert = require('assert');
-const werelogs = require('werelogs');
 
+const HttpAgent = require('agentkeepalive');
+const HttpsAgent = require('agentkeepalive').HttpsAgent;
```
Can this be?
```suggestion
const HttpsAgent = HttpAgent.HttpsAgent;
```

jonathan-gramain replied:
I don't know, since I don't have a https setup to check and rely on Integration for https validation, and since the module documentation uses this form, I'd rather leave it as is in doubt. I haven't seen this way of nesting before but it's possible that there is special handling by the module when doing a simple `HttpAgent = require(..)` to return the appropriate class.

miniscruff replied:
Ok fair, some of the require / import things in JS are still a little fuzzy to me.

jonathan-gramain commented:
ping

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue S3C-3114 contains:


* `7.4.7`

* `7.7.1`

* `7.8.0`


Considering where you are trying to merge, I expected to find:


* `7.4.7`

* `7.7.1`

* `7.8.0`

* `8.1.0`


Please check the `Fix Version/s` of S3C-3114, or the target
branch of this pull request.

jonathan-gramain commented:
Will do a bypass_jira_check because currently there is no way to include version 8.1.0 in the S3C ticket, and the relaxing of the check rule is not yet activated.

jonathan-gramain commented:
/bypass_jira_check
/approve

bert-e commented:
# Not authorized



I'm afraid I cannot do that, @jonathan-gramain:

> /bypass_jira_check
> /approve


You cannot use `bypass_jira_check` in your own pull request.


Please **edit** or **delete** the corresponding comment so I can move on.

jonathan-gramain commented:
/approve

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue S3C-3114 contains:


* `7.4.7`

* `7.7.1`

* `7.8.0`


Considering where you are trying to merge, I expected to find:


* `7.4.7`

* `7.7.1`

* `7.8.0`

* `8.1.0`


Please check the `Fix Version/s` of S3C-3114, or the target
branch of this pull request.




*The following options are set:* **approve**

rahulreddy commented:
/bypass_jira_check

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.1/bugfix/S3C-3114-freeSocketTimeoutInAgent` with contents from `w/7.8/bugfix/S3C-3114-freeSocketTimeoutInAgent`
and `development/8.1`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.1/bugfix/S3C-3114-freeSocketTimeoutInAgent origin/development/8.1
 $ git merge origin/w/7.8/bugfix/S3C-3114-freeSocketTimeoutInAgent
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.1/bugfix/S3C-3114-freeSocketTimeoutInAgent
```





*The following options are set:* **approve, bypass_jira_check**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/S3C-3114-freeSocketTimeoutInAgent` into
`development/7.4`
* [w/7.7/bugfix/S3C-3114-freeSocketTimeoutInAgent](https://github.com/scality/bucketclient/compare/development/7.7...w/7.7/bugfix/S3C-3114-freeSocketTimeoutInAgent) will be merged into `development/7.7` (pull request #188)
* [w/7.8/bugfix/S3C-3114-freeSocketTimeoutInAgent](https://github.com/scality/bucketclient/compare/development/7.8...w/7.8/bugfix/S3C-3114-freeSocketTimeoutInAgent) will be merged into `development/7.8` (pull request #189)
* [w/8.1/bugfix/S3C-3114-freeSocketTimeoutInAgent](https://github.com/scality/bucketclient/compare/development/8.1...w/8.1/bugfix/S3C-3114-freeSocketTimeoutInAgent) will be merged into `development/8.1` (pull request #190)



The following branches will **NOT** be impacted:

* `development/6.4`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.







*The following options are set:* **approve, bypass_jira_check**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/7.4`

* :heavy_check_mark: `development/7.7`

* :heavy_check_mark: `development/7.8`

* :heavy_check_mark: `development/8.1`



The following branches will **NOT** be impacted:

* `development/6.4`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve, bypass_jira_check**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/7.4`

* :heavy_check_mark: `development/7.7`

* :heavy_check_mark: `development/7.8`

* :heavy_check_mark: `development/8.1`



The following branches have **NOT** changed:

* `development/6.4`



Please check the status of the associated issue S3C-3114.

Goodbye jonathan-gramain.
